presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -775,6 +789,13 @@ protected boolean internalStartNext()
             }
             boolean started = subGroup.internalStartNext();
             checkState(started, "Eligible sub group had no queries to run");
+
+            long currentTime = System.currentTimeMillis();
+            if (lastStartMillis != 0) {
```

Comment:
I don't get this. So when the resource group becomes ineligible, `lastStartMillis` will be reset to 0, so you won't update `timeBetweenStartsSec` when the first time this group is scheduled after becoming eligible again, but only update for the consecutive new queries scheduled after the first one. Am I understanding it right? What is this useful for?

Reply:
Correct, it measures time between eligible submissions getting run.

Reply:
What will this be used for?